fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,189 @@
+        )
+
+        expect:
+        outerMetricUnionAvailability.getDataSourceNames() == [TableName.of('source1'), TableName.of('source2')] as Set
+    }
+
+    @Unroll
+    def "checkDuplicateValue returns duplicates of metric column names from 2 physical tables in the case of #caseDescription"() {
+        when:
+        Map<MetricColumn, Set<Availability>> actual = MetricUnionAvailability.getDuplicateValue(
+                [
+                        (availability1): metricColumnNames1.collect{it -> new MetricColumn(it)} as Set,
+                        (availability2): metricColumnNames2.collect{it -> new MetricColumn(it)} as Set
+                ]
+        )
+
+        then:
+        actual.size() == expected.size()
+        for (Map.Entry<String, Integer> entry : expected.entrySet()) {
+            MetricColumn duplicate = new MetricColumn(entry.key)
+            actual.containsKey(duplicate)
+            actual.get(duplicate).size() == entry.value
+        }
+
+        where:
+        metricColumnNames1     | metricColumnNames2 | expected       | caseDescription
+        []                     | []                 | [:]            | 'both physical tables having empty metric column names'
+        ['metric1']            | []                 | [:]            | 'one empty table and one non-empty table'
+        ['metric1']            | ['metric1']        | ['metric1': 2] | 'both physical tables having the same metric column names'
+        ['metric1', 'metric2'] | ['metric1']        | ['metric1': 2] | 'two tables having intersections'
+        ['metric1']            | ['metric2']        | [:]            | 'two tables having no intersections'
+    }
+
+    def "getAllAvailableIntervals returns configured metric columns with their corresponding available intervals in union"() {
+        given:
+        availability1.getAllAvailableIntervals() >> [
+                (new MetricColumn('metric1')): [interval1] as List,
+                (new MetricColumn('metric3')): [interval3] as List
+        ]
+        availability1.getDataSourceNames() >> Sets.newHashSet(TableName.of('source1'))
+        availability2.getAllAvailableIntervals() >> [
+                (new MetricColumn('metric1')): [interval4] as List,
+                (new MetricColumn('metric2')): [interval2] as List
+        ]
+        availability2.getDataSourceNames() >> Sets.newHashSet(TableName.of('source2'))
+
+        schema1.getColumns(_) >> Collections.emptySet()
+        schema2.getColumns(_) >> Collections.emptySet()
+
+        metricUnionAvailability = new MetricUnionAvailability(
+                [physicalTable1, physicalTable2] as Set,
+                [metricColumn1, metricColumn2] as Set
+        )
+
+        expect:
+        metricUnionAvailability.getAllAvailableIntervals() == [
+                (metricColumn1): [interval1, interval4] as List,
+                (metricColumn2): [interval2] as List,
+        ]
+    }
+
+    def "test getAvailableIntervals"() {
+        given:
```

Comment:
This test tests only a path with no overlap.  Would it make sense to also test when there is an overlap?